Makefile

```diff
@@ -1,4 +1,5 @@
-LUAJIT_O := deps/luajit/src/libluajit.a
+ABS_TOP_SRCDIR=`pwd`
```

Comment:
Could we do this in all files:

TOP_SRCDIR:=<whatever>
include $(TOP_SRCDIR)/common.mk

and in common.mk

ABS_TOP_SRCDIR:=$(shell cd $(TOP_SRCDIR) && pwd)

Additionally in common.mk we could define LUAJIT as the full path to luajit, and that will eventually let us remove all PATH manipulations in our makefiles as we can just reference binaries by their absolute paths using convenient make variables.  WDYT?